beacon-chain/core/transition/transition.go

```diff
@@ -257,16 +257,18 @@ func ProcessSlots(ctx context.Context, state state.BeaconState, slot primitives.
 					return nil, errors.Wrap(err, "could not process epoch with optimizations")
 				}
 			} else if state.Version() <= version.Deneb {
+				stateVersion := version.String(state.Version())
```

Comment:
maybe we should do it outside , but it's also possible to trigger the nil pointer issue in the else statement maybe. It was my idea to add this state version but seems to be causing other issues.